common/djangoapps/entitlements/models.py

```diff
@@ -27,35 +27,59 @@ class CourseEntitlement(TimeStampedModel):
     enrollment_course_run = models.ForeignKey('student.CourseEnrollment', null=True)
     order_number = models.CharField(max_length=128, null=True)
 
+    def to_json(self):
+        run_key = None
+
+        if self.enrollment_course_run is not None:
+            run_key = self.enrollment_course_run.course_id
+
+        return {
+            'username': self.user.username,
+            'uuid': str(self.uuid),
+            'created': str(self.created),
+            'modified': str(self.modified),
+            'course_uuid': str(self.course_uuid),
+            'expired_at': str(self.expired_at),
+            'mode': self.mode,
+            'enrollment_course_run_key': run_key,
+            'order_number': self.order_number,
+        }
+
     @classmethod
-    def entitlements_for_user(cls, user):
+    def get_user_course_entitlement(cls, user, course_uuid, ignore_expired=True):
         """
-        Retrieve all the Entitlements for a User
+        Retrieve The entitlement for the given parent course id if it exists for the User and is active
```

Comment:
What does it mean for the entitlement to be active? Is this different from being expired? How are you determining that the entitlement is active?
(Sorry if the three successive questions seems intense; I just want to convey all the things I'm wondering)